lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,10 +71,16 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
-          if ActiveModel::Serializer.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
+          unless serializer.class._type.nil?
+            serializer.class._type
           else
-            serializer.object.class.model_name.plural
+            if ActiveModel::Serializer.config.type_source == :model
```

Comment:
I think this should be extracted to another PR. I'm generally not in favor of adding too many config options, and with the `type` method you introduced, there's virtually no use for this one, except for allowing wildly different workflows for users, which, as well, I'm not too keen on.